### ISSUE TITLE
enhancement: a11y rule "label-has-associated-control" fau

### ISSUE THREAD

**Issue by FatumaA** [open-sauced/app#4126]
### Describe the bug

[ jsx-a11y eslint plugin](https://www.npmjs.com/package/eslint-plugin-jsx-a11y) was recently added to the project and this rule - jsx-a11y/label-has-associated-control - fails with a warning. 
 

This issue offers more context: #2264 

### Steps to reproduce

The following components and files are causing the rule to fail: 
1. components/AddContributorsHeader/add-contributors-header.tsx
2. components/organisms/UserSettingsPage/user-settings-page.tsx
3. components/shared/AppSidebar/AppSidebar.tsx
4. components/shared/DayRangePicker.tsx
5. components/shared/LimitPicker.tsx

**Comment by shubhamchasing** [comment]
.take
